runtime/parachains/src/paras.rs

```diff
@@ -826,6 +858,36 @@ impl<T: Config> Module<T> {
 		shared::Module::<T>::scheduled_session()
 	}
 
+	/// Store the validation code if not already stored, and increase the number of reference.
+	///
+	/// Returns the hash, number of storage reads and number of storage writes.
+	fn increase_code_ref(validation_code: &ValidationCode) -> (Hash, u64, u64) {
```

Comment:
These functions should take `&Hash` instead of the entire validation code. That would go well with the changes to `PastCode`, `CurrentCode`, and `FutureCode`.